src/views/ActivityLogView.vue

```diff
@@ -1,5 +1,264 @@
+            <input type="date" v-model="date" />
+          </div>
+        </div>
+        <div class="action-buttons">
+          <button @click="resetFilters">Clear Filters</button>
+          <button @click="applyFilters" class="action-buttons__filter-btn">
+            <img src="@/assets/icons/filter.svg" alt="filter-icon" /> Filter
+          </button>
+        </div>
+      </div>
+    </div>
+    <div class="events">
+      <div class="events__header">
+        <h2 class="title">Events</h2>
+        <BaseSearch v-model="searchTerm" placeholder="Search..." />
+      </div>
+      <EventsTable
+        ref="eventsTable"
+        :searchTerm="searchTerm"
+        :filters="activeFilters"
+      />
+    </div>
   </div>
 </template>
+
+<script>
+import angleDownImage from '@/assets/icons/angle-down.svg'
+import angleUpImage from '@/assets/icons/angle-up.svg'
+import BaseSelect from '@/components/BaseSelect.vue'
+import { DatePicker } from 'primevue'
+import BaseSearch from '@/components/BaseSearch.vue'
+import EventsTable from '@/components/EventsTable/EventsTable.vue'
+
+export default {
+  name: 'ActivityLogView',
+  components: { BaseSelect, DatePicker, BaseSearch, EventsTable },
+  data() {
+    return {
+      selectModelOptions: [
+        { name: 'Model', value: 'Model' },
+        { name: 'Success', value: 0 },
+        { name: 'Update', value: 1 },
+        { name: 'Carrier notification', value: 2 },
+      ],
+      selectModelValue: 'Model',
+      selectConditionOptions: [
+        { name: 'Condition', value: 'Condition' },
+        { name: 'Question', value: 0 },
+        { name: 'Action', value: 1 },
+        { name: 'Information', value: 2 },
+        { name: 'Suggestion', value: 3 },
+        { name: 'Agreement', value: 4 },
+      ],
+      selectConditionValue: 'Condition',
+      location: '',
+      date: '',
+      angleDownImage: '@assets/icons/angle-down.svg',
+      angleUpImage: '@assets/icons/angle-up.svg',
+      showFilters: false,
+      searchTerm: '',
+      activeFilters: {},
+      locationError: '',
+    }
+  },
+  computed: {
+    filterIcon() {
+      return this.showFilters ? angleUpImage : angleDownImage
+    },
+  },
+  methods: {
+    toggleFilters() {
+      this.showFilters = !this.showFilters
+    },
+    handleSelectModelOption(option) {
+      this.selectModelValue = option.name
+      this.searchTerm = ''
+    },
+    handleSelectConditionOption(option) {
+      this.selectConditionValue = option.name
+      this.searchTerm = ''
+    },
+    resetFilters() {
+      this.selectModelValue = 'Model'
+      this.selectConditionValue = 'Condition'
+      this.location = ''
+      this.date = ''
+      this.applyFilters()
+    },
+    applyFilters() {
+      this.activeFilters = {
+        model: this.selectModelValue,
+        condition: this.selectConditionValue,
+        location: this.location,
+        date: this.date,
+      }
+      this.$refs.eventsTable.applyFilters(this.activeFilters)
```

Comment:
this is code seem to produce an error
<img width="412" alt="image" src="https://github.com/user-attachments/assets/31d194ef-e469-428a-9be7-a73362bf6321">

Even if it wasn't causing error, this is not recommended approach in this particular case. Wht you should do instead is to have two filter states: one in the filters component, the other in this component. As soon as you press 'apply filters' button you should emit filters state up and then pass it to the table as a prop. This would be a more intuitive data flow.

Refs are usually used when we want to trigger some layout side effect. Although refs can be used for business logic, it makes data flow more cumbersome and is only allowed as a temporary crutch until a better solution will be implemented.

Reply:
It seems like I forgot to remove that line with refs, because I havy already passed the filter values to table)